### PR TITLE
feat(config): only warn while `experiments.newSplitChunks` being explicitly setted

### PR DESCRIPTION
**Pull request by hyf0**
## Summary

Fix https://github.com/web-infra-dev/rspack/pull/4169#issuecomment-1715039630

## Require Documentation?

- [ ] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Review by ahabhgk — DISMISSED** [review]

File: packages/rspack/src/config/zod.ts [hunk 1]
```diff
@@ -944,10 +944,12 @@ const experiments = z.strictObject({
 	newSplitChunks: z
 		.boolean()
 		.optional()
-		.refine(_ => {
-			console.warn(
-				"`experiments.newSplitChunks` will be removed at 0.4.0. See details at https://github.com/web-infra-dev/rspack/discussions/4168"
-			);
+		.refine(val => {
+			if (val === false) {
```
Comment (h-a-n-a) [inline packages/rspack/src/config/zod.ts:948]: Maybe we should warn every user that sets this value explicitly no matter `true`/`false` is used?

Comment (h-a-n-a) [reply packages/rspack/src/config/zod.ts:948]: Is it possible to get the raw user input here?

Comment (h-a-n-a) [reply packages/rspack/src/config/zod.ts:948]: According to the deprecation message "`experiments.newSplitChunks` will be removed at 0.4.0.", looks like users with option `true` needs to be warned too.

Comment (hyf0) [reply packages/rspack/src/config/zod.ts:948]: the `val` should be the raw user input.

**Review by h-a-n-a — APPROVED** [review]

**Comment by h-a-n-a** [comment]
LGTM

**Pull request merged**